testplan/runners/pools/base.py

```diff
@@ -799,15 +611,16 @@ def _add_workers(self):
             worker.parent = self
             worker.cfg.parent = self.cfg
             self._workers.add(worker, uid=idx)
+            self._conn.register(worker)
             self.logger.debug('Added worker %(index)s (outfile = %(outfile)s)',
                               {'index': idx, 'outfile': worker.outfile})
 
     def starting(self):
         """Starting the pool and workers."""
         with self._pool_lock:
             self._conn.start()
-            for worker in self._workers:
-                self._conn.register(worker)
+            if not len(self._workers):
```

Comment:
It feels a bit too scholastic to have to add all workers again if we restart the pool, and use case is limited in our own test. I think this would do, for now.